connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectPlannerSuite.scala

```diff
@@ -571,4 +572,28 @@ class SparkConnectPlannerSuite extends SparkFunSuite with SparkConnectPlanTest {
       Dataset.ofRows(spark, transform(proto.Relation.newBuilder.setProject(project).build()))
     assert(df.schema.fields.toSeq.map(_.name) == Seq("id"))
   }
+
+  test("Test Hint") {
```

Comment:
Maybe also add a test case to have a random hint name? I expect Catalyst will throw something.